BootloaderCorePkg/Library/MpInitLib/Ia32/MpFuncs.nasm

```diff
@@ -70,13 +83,19 @@ FLAT32_JUMP:
   dd 0h                         ; 32-bit offset
   dw 0h                         ; 16-bit selector
 
-ProtModeStart:                 ; protected mode entry point
+SmmRebase:
+  mov         edi,  0x3fef8     ; SMBASE offset
+  mov         dword [edi], eax  ; change to new  SMBASE
+  mov         dword [eax + 0x8000], 0x9090AA0F  ; write 'RSM' at new SMM handler entry
+  rsm
+  jmp         $-2
```

Comment:
This is for infinite rsm call just in case?

Reply:
This is to change the default SMBASE from 0x30000 to new location in TSEG.   And at the new SMM entry location, we put a "RSM" as the 1st instruction.

Reply:
My question is about a line - jmp $-2. It shouldn't be reachable.